effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -580,6 +606,26 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
       val (exprs, strs) = manyWhile((`${` ~> expr() <~ `}`, string()), `${`).unzip
       Template(first :: strs, exprs)
 
+  def documented[T](p: Doc => T): T =
+    p(maybeDocumentation())
+
+  def maybeDocumentation(): Doc =
+    nonterminal:
+      peek.kind match
+        case DocComment(msg) =>
+          consume(peek.kind)
+          val next = maybeDocumentation()
+          next.map { d => msg ++ "\\n" ++ d }
+            .orElse(Some(msg))
```

Comment:
Again, I am afraid that computing this every time (even if we do not generate docs) might be slow. Also this should really be a tight loop using a string buffer (like `infix`).